src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,1613 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/mocks/ERC1271WalletMock.sol";
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetFixedSupply.sol";
+
+import "src/contracts/core/PaymentCoordinator.sol";
+import "src/contracts/strategies/StrategyBase.sol";
+
+import "src/test/events/IPaymentCoordinatorEvents.sol";
+import "src/test/utils/EigenLayerUnitTestSetup.sol";
+import "src/test/mocks/Reenterer.sol";
+import "src/test/mocks/ERC20Mock.sol";
+
+/**
+ * @notice Unit testing of the PaymentCoordinator contract
+ * Contracts tested: PaymentCoordinator
+ * Contracts not mocked: StrategyBase, PauserRegistry
+ */
+contract PaymentCoordinatorUnitTests is EigenLayerUnitTestSetup, IPaymentCoordinatorEvents {
+    // used for stack too deep
+    struct FuzzPayForRange {
+        address avs;
+        bool retroactivePaymentsEnabled;
+        uint256 startTimestamp;
+        uint256 duration;
+        uint256 amount;
+    }
+
+    // Contract under test
+    PaymentCoordinator public paymentCoordinator;
+    PaymentCoordinator public paymentCoordinatorImplementation;
+
+    // Mocks
+    IERC20 token1;
+    IERC20 token2;
+    IERC20 token3;
+    IStrategy strategyMock1;
+    IStrategy strategyMock2;
+    IStrategy strategyMock3;
+    StrategyBase strategyImplementation;
+    uint256 mockTokenInitialSupply = 10e50;
+    IPaymentCoordinator.StrategyAndMultiplier[] defaultStrategyAndMultipliers;
+
+    // Config Variables
+    /// @notice Max duration is 5 epochs (2 weeks * 5 = 10 weeks in seconds)
+    uint64 MAX_PAYMENT_DURATION = 86400 * 70;
+
+    /// @notice Lower bound start range is 3 months into the past
+    uint64 MAX_RETROACTIVE_LENGTH = 86400 * 90;
```

Comment:
you can do things like `90 days`

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/488/commits/a27911bceb4c60db7ad1a53551301dfb4615d020
Cleaned up